prometheus_client/samples.py

```diff
@@ -48,6 +65,6 @@ class Exemplar(NamedTuple):
 class Sample(NamedTuple):
     name: str
     labels: Dict[str, str]
-    value: float
+    value: Union[float, NativeHistStructValue]
```

Comment:
Trying to decide if this should be a union or if we should have a separate field for `histogram_value` or something like that.

Reply:
Given the union's "side-effects" in files like `registry` etc, maybe it should be a separate field? My initial thinking with the union was to possibly avoid function signature changes I guess, but maybe it wouldn't be that problematic?

Reply:
Yeah, let's try having a separate Optional field for the native histogram and see what it looks like. It also makes it nice to just check if the value is none or not for if a value is a native histogram.

Reply:
I separated them, and added a field for native histogram. I added it for last to minimize the risk of breaking something.